lib/core.ps1

```diff
@@ -507,6 +510,7 @@ function Invoke-ExternalCommand {
     }
     $Process = New-Object System.Diagnostics.Process
     $Process.StartInfo.FileName = $FilePath
+    $Process.StartInfo.CreateNoWindow = $NoWindow
```

Comment:
Move this line to L523/L527's `if`, and should this flag be set to true by default? Since normally we don't need a new window here.

Reply:
> should this flag be set to true by default? Since normally we don't need a new window here.

No, it doesn't have to be. Mostly we use this function to invoke CLI programs, so `-NoWindow` is just for any other that has a GUI.

Reply:
```pwsh
    if ($RunAs) {
        $Process.StartInfo.UseShellExecute = $true
        $Process.StartInfo.Verb = 'RunAs'
    } else {
        $Process.StartInfo.CreateNoWindow = $true
    }
```

Only if `Verb = 'RunAs'` `CreateNoWindow` will not be used, and is this a better solution?

Reply:
Oh, yes. Now I got what you meant. ✅

Reply:
> No, it doesn't have to be. Mostly we use this function to invoke CLI programs, so -NoWindow is just for any other that has a GUI.

Now this one... Just in my code above, set `CreateNoWindow` to true will not add costs to CLI process and for GUI one, we always don't want a new window if run in `Invoke-ExternalCommand`, right?

Reply:
> for GUI one, we always don't want a new window if run in `Invoke-ExternalCommand`, right?

Should we keep a way for possible user interaction during the installation? Although this may be rare. I definitely understand and agree that "no GUI" is part of Scoop's philosophy.

Reply:
No, we don't need to keep such a way, so do installation arguments, IMO.

For different installation types, using customized manifest should be better.

Reply:
Fine. Then we should remove this switch and set `CreateNoWindow` to `$true`.